hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -605,6 +605,18 @@ object DataSourceWriteOptions {
 
   val DROP_PARTITION_COLUMNS: ConfigProperty[java.lang.Boolean] = HoodieTableConfig.DROP_PARTITION_COLUMNS
 
+  val ENABLE_OPTIMIZED_UPDATE: ConfigProperty[String] = ConfigProperty
+    .key("hoodie.enable.spark.sql.optimized.update")
+    .defaultValue("true")
+    .markAdvanced()
+    .withDocumentation("Controls whether spark sql optimized update is enabled.")
+
+  val ENABLE_OPTIMIZED_DELETE: ConfigProperty[String] = ConfigProperty
+    .key("hoodie.enable.spark.sql.optimized.delete")
```

Comment:
lets try to follow our usual way of naming configs. generally we try to keep "enable" in the end. 
"hoodie.spark.sql.writes.optimized.updates.enable"

Reply:
Changed to `hoodie.spark.sql.writes.optimized.enable`